### ISSUE TITLE
Services that come up in new pods are unable to resolve DNS and connect to RDS

### ISSUE THREAD

**Issue by powerupcloud** [kubernetes/kubernetes#43752]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.): AWS, DNS, RDS

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Bug Report

**Kubernetes version** (use `kubectl version`): 1.4.4

**Environment**:
- **Cloud provider or hardware configuration**: AWS
- **OS** (e.g. from /etc/os-release): Debian 8
- **Kernel** (e.g. `uname -a`): 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux
- **Install tools**: KubeDNS and other standard tools
- **Others**:

**What happened**: 
Whenever a new pod comes up the service is not able to resolve DNS and fails connecting to other AWS services like RDS. Usual suspects like security group rules etc to RDS are all in place and can be ruled out. We then terminate the pod and launch a new one - the new pod too makes multiple connection attempts before finally being able to connect to RDS. Please take a look at the log below, captured during one of these failures. 

![2017-03-28](https://cloud.githubusercontent.com/assets/12628600/24405897/165e6f54-13e4-11e7-81d5-4cffd533cefd.png)

**What you expected to happen**:
That the new pod comes up and services are able to connect to RDS without failing due to DNS resolution. 

**How to reproduce it** (as minimally and precisely as possible):
We haven't noticed a specific way to reproduce this. This happens sporadically in our environment bringing services down. 

**Anything else we need to know**:

**Comment by powerupcloud** [comment]
More data. When this problem occurs, its not just the resolution to RDS that fails, but even simple requests like looking up google.com also fail.

**Comment by adamneilson** [comment]
We're seeing a very similar issue accessing RDS from minions but do not see the same problem from the master. We have a peering connection from k8s cluster VPC `172.20.0.0/16` to RDS VPC `172.31.0.0/16`.

`telnet my-database.eu-west-1.rds.amazonaws.com 3306` from master gives the `mysql_native_password` challenge but running that same command from within a container on a minion times out.

Checked DNS and tried to telnet directly to the database IP but that times out also which suggests its not a DNS issue.

Up until today there was a weird workaround: When I deployed a new version and the pod was created the connection to RDS timed out. If I deleted that pod and let the replication controller start a new pod, that pod would then connect. This worked up until this afternoon.

```
$]› kubectl version
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"2016-12-14T00:57:05Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T04:52:34Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```
